trace/trace.go

```diff
@@ -0,0 +1,348 @@
+package trace
+
+import (
+	"context"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/honeycombio/beeline-go/propagation"
+	"github.com/honeycombio/beeline-go/sample"
+	"github.com/honeycombio/beeline-go/timer"
+	libhoney "github.com/honeycombio/libhoney-go"
+)
+
+var GlobalConfig Config
+
+type Config struct {
+	// TODO describe what these are and the return values
+	SamplerHook func(map[string]interface{}) (bool, int)
+	PresendHook func(map[string]interface{})
+}
+
+// Trace holds some trace level state and the root of the span tree that will be
+// the entire in-process trace. Traces are sent to Honeycomb when the root span
+// is finished. You can send a trace manually, and that will cause all
+// synchronous  spans in the trace to be finished and sent. Asynchronous spans
+// must still be sent manually
+type Trace struct {
+	// TODO add shouldDrop
```

Comment:
TODO?

Reply:
I think premature optimization to think about `shouldDrop`. Removed the comment.